monitor/data_monitoring.py

```diff
@@ -64,43 +80,68 @@ def _update_sent_alerts(self, alert_ids) -> None:
         test_result_alerts = []
+        failed_to_parse_dict_alerts = []
         if results:
             test_result_alert_dicts = json.loads(results[0])
             self.execution_properties['alert_rows'] = len(test_result_alert_dicts)
             for test_result_alert_dict in test_result_alert_dicts:
-                test_result_object = TestResult.create_test_result_from_dict(
-                    test_result_dict=test_result_alert_dict,
-                )
-                if test_result_object:
-                    test_result_alerts.append(test_result_object)
+                test_result = TestResult.create_test_result_from_dict(test_result_alert_dict)
+                if test_result:
+                    test_result_alerts.append(test_result)
                 else:
+                    failed_to_parse_dict_alerts.append(test_result_alert_dict)
                     self.success = False
 
-        return test_result_alerts
+        return AlertsQueryResult(test_result_alerts, failed_to_parse_dict_alerts)
 
-    def _send_to_slack(self, test_result_alerts: List[TestResult]) -> None:
+    def _send_alerts_to_slack(self, alerts: List[SlackAlert]) -> None:
         sent_alerts = []
-        alerts_with_progress_bar = alive_it(test_result_alerts, title="Sending alerts")
+        alerts_with_progress_bar = alive_it(alerts, title="Sending alerts")
         for alert in alerts_with_progress_bar:
-            alert_slack_message: SlackMessageSchema = alert.generate_slack_message(is_slack_workflow=self.config.is_slack_workflow)
             sent_successfully = self.slack_client.send_message(
                 channel_name=self.slack_channel_name,
-                message=alert_slack_message
+                message=alert.message
             )
             if sent_successfully:
                 sent_alerts.append(alert.id)
             else:
-                logger.error(f"Could not sent the alert - {alert.id}. Full alert: {json.dumps(dict(alert_slack_message))}")
+                logger.error(
+                    f"Could not sent the alert - {alert.id}. Full alert: {json.dumps(dict(alert.message))}")
                 self.success = False
-        
+
         sent_alert_count = len(sent_alerts)
         self.execution_properties['sent_alert_count'] = sent_alert_count
         if sent_alert_count > 0:
             self._update_sent_alerts(sent_alerts)
 
+    def _get_slack_alert_from_test_result(self, test_results: List[TestResult]):
+        return [
+            SlackAlert(
+                test_result.id,
+                test_result.generate_slack_message(is_slack_workflow=self.config.is_slack_workflow)
+            )
+            for test_result in test_results
+        ]
+
+    @staticmethod
+    def _get_slack_alert_from_dict(alert_dicts: List[Dict[str, Any]]):
+        return [
+            SlackAlert(
+                alert_dict['id'],
```

Comment:
Use get here? (I know it should always be there but just in case as this is the last resort :))

Reply:
I can't, the entire logic would break if an alert doesn't have an alert ID.  
How would we flag that it was sent?